File: mpi_examples/fortran_c_codes/example1/example1_mpi.f90
program mpi_example1 
use mpi 
implicit none 
integer ierr, np, rank, inum

call MPI_INIT(ierr) 
call MPI_COMM_SIZE(MPI_COMM_WORLD, np, ierr)
call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr) 

if (rank == 0) then
write(*,*) 'I am master with rank ',rank
else 
write(*,*) 'My rank is: ',rank
endif 

if (rank == 0) write(*,*) np

call MPI_Finalize(ierr) 

end program mpi_example1

File: mpi_examples/fortran_c_codes/example1/example1_mpi.c
#include <mpi.h>
#include <stdio.h>
int main(int argc, char **argv){
int np, rank;

MPI_Init(&argc, &argv);
MPI_Comm_size(MPI_COMM_WORLD, &np);
MPI_Comm_rank(MPI_COMM_WORLD, &rank);

if (rank == 0){
printf("I am master with rank: %d\n",rank);
}
else {
printf("My rank is: %d\n",rank);
}

MPI_Finalize();
}
